src/common/safe_int.ts

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export function toSafeInt(n: number, def?: number) {
+    if (Number.isSafeInteger(n)) {
```

Comment:
:neckbeard: Should we also add a simple `typeof` check here as well? Even though this is TS, it is getting called from JS at the moment. You are not assured that you are getting a `number`. For example, this will accept `boolean` and empty arrays as the `n` value without triggering the default value or throwing.